charts/cryostat/templates/deployment.yaml

```diff
@@ -34,69 +36,61 @@ spec:
+          - name: QUARKUS_HTTP_PROXY_ALLOW_X_FORWARDED
+            value: 'true'
+          - name: QUARKUS_HTTP_PROXY_ENABLE_FORWARDED_HOST
+            value: 'true'
+          - name: QUARKUS_HTTP_PROXY_ENABLE_FORWARDED_PREFIX
+            value: 'true'
+          - name: QUARKUS_HIBERNATE_ORM_DATABASE_GENERATION
+            value: drop-and-create
+          - name: QUARKUS_DATASOURCE_USERNAME
+            value: cryostat3
+          - name: QUARKUS_DATASOURCE_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ printf "%s-db-connection-key" .Release.Name }}
+                key: CONNECTION_KEY
+                optional: false
+          - name: QUARKUS_DATASOURCE_JDBC_URL
+            value: jdbc:postgresql://localhost:5432/cryostat3
+          - name: STORAGE_BUCKETS_ARCHIVES_NAME
+            value: archivedrecordings
+          - name: QUARKUS_S3_ENDPOINT_OVERRIDE
+            value: http://localhost:8333
+          - name: QUARKUS_S3_PATH_STYLE_ACCESS
             value: "true"
-          {{- end }}
-          {{- if ((.Values.pvc).enabled) }}
-          - name: CRYOSTAT_JDBC_URL
-            value: jdbc:h2:file:/opt/cryostat.d/conf.d/h2;INIT=create domain if not exists jsonb as varchar
-          - name: CRYOSTAT_HBM2DDL
-            value: update
-          - name: CRYOSTAT_JDBC_DRIVER
-            value: org.h2.Driver
-          - name: CRYOSTAT_HIBERNATE_DIALECT
-            value: org.hibernate.dialect.H2Dialect
-          - name: CRYOSTAT_JDBC_USERNAME
-            value: {{ include "cryostat.fullname" . }}
-          - name: CRYOSTAT_JDBC_PASSWORD
-            value: {{ include "cryostat.fullname" . }}
-          {{- end }}
-          - name: CRYOSTAT_JMX_CREDENTIALS_DB_PASSWORD
+          - name: QUARKUS_S3_AWS_REGION
+            value: us-east-1
+          - name: QUARKUS_S3_AWS_CREDENTIALS_TYPE
+            value: static
+          - name: QUARKUS_S3_AWS_CREDENTIALS_STATIC_PROVIDER_ACCESS_KEY_ID
+            value: cryostat
+          - name: AWS_ACCESS_KEY_ID
+            value: cryostat
+          - name: QUARKUS_S3_AWS_CREDENTIALS_STATIC_PROVIDER_SECRET_ACCESS_KEY
             valueFrom:
               secretKeyRef:
-                name: {{ default (printf "%s-jmx-credentials-db" .Release.Name) .Values.core.databaseSecretName }}
-                key: CRYOSTAT_JMX_CREDENTIALS_DB_PASSWORD
+                name: {{ printf "%s-storage-secret-key" .Release.Name }}
+                key: SECRET_KEY
                 optional: false
-          - name: CRYOSTAT_AUTH_MANAGER
-          {{- if (.Values.authentication).basicAuth.enabled }}
-            value: io.cryostat.net.BasicAuthManager
-          {{- else }}
-            value: io.cryostat.net.NoopAuthManager
+          - name: AWS_SECRET_ACCESS_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ printf "%s-storage-secret-key" .Release.Name }}
+                key: SECRET_KEY
+                optional: false
```

Comment:
For my own info, why are there two pairs of these environment variables? (e.g. `QUARKUS_S3_AWS_CREDENTIALS_STATIC_PROVIDER_SECRET_ACCESS_KEY` and `AWS_SECRET_ACCESS_KEY`).

Reply:
I need to double-check that myself, actually. I think the `QUARKUS_` one is what is actually needed on the Cryostat side, and the `AWS_` one was needed for Minio previously? Or perhaps the SDK client does actually need both, for some reason.

Reply:
I think this was probably just a configuration hold-over:

https://docs.quarkiverse.io/quarkus-amazon-services/dev/amazon-s3.html#quarkus-amazon-s3_quarkus.s3.aws.credentials.type

If using the `env-variable` credentials type then the `AWS_SECRET_ACCESS_KEY` would be checked. Since in any case this configuration is going to expose the credentials to the container via environment variables, just using `static` with the `QUARKUS_`-prefixed pair is simpler.

Reply:
Looks like this was actually used for something after all: https://github.com/cryostatio/cryostat-helm/issues/120